app/components/Header/Header.js

```diff
@@ -1,29 +1,75 @@
-import React from 'react'
+import React, { Component, PropTypes } from 'react'
+import { connect } from 'react-redux'
+import { memberInfoFetchData, logoutUserAction } from '../../actions'
 import './Header.scss'
 
-export default class Header extends React.Component {
-  render() {
-    return (
-  		<section className="marketplace--header">
-      		<div className="wrapper">
-		        <div className="marketplace--logo">
-          			<a href="/" title="Go to the Marketplace homepage" className="logo">Digital Marketplace</a> <span className="badge--beta">BETA</span>
-		        </div>
-		        <div className="user-nav">    
-	          		<div id="react-bundle-auth-header-state"></div>
-					<div id="react-bundle-auth-header">
-						<ul data-reactroot="" id="main-navigation" className="inline-links--inverted">
-							<li>
-								<a href="/signup">Join the Marketplace</a>
-							</li>
-							<li>
-								<a href="/login?next=/signup">Sign in</a>
-							</li>
-						</ul>
-					</div>
-		        </div>
-      		</div>
-    	</section>
-    );
-  }
-}
\ No newline at end of file
+class Header extends Component {
+	componentWillMount () {
+	    this.props.fetchData(this.props.userSessionCookie)
+	}
+
+	 dashBoardLink = () => {
+	    if (this.props.memberInfo.userType == 'buyer') {
+	      return <a href="/buyers">Dashboard</a>
+	    } else if (this.props.memberInfo.userType == 'applicant') {
+	    	return <a href="/sellers/application">Continue application</a>
+	    } else {
+	      return <a href="/sellers">Dashboard</a>
+	    }
+  	}
+
+	render() {
+	    return (
+	  		<section className="marketplace--header">
+	      		<div className="wrapper">
+			        <div className="marketplace--logo">
+	          			<a href="/" title="Go to the Marketplace homepage" className="logo">Digital Marketplace</a> <span className="badge--beta">BETA</span>
+			        </div>
+			        <div className="user-nav">    
+		          		<div id="react-bundle-auth-header-state"></div>
+						<div id="react-bundle-auth-header">
+							<ul data-reactroot="" id="main-navigation" className="inline-links--inverted">
+								<li>
+									{this.props.memberInfo.isAuthenticated ?
+										<span>{ this.dashBoardLink() }</span>
+										:
+										<a href="/signup">Join the Marketplace</a>
+									}
+								</li>
+								<li>
+									{this.props.memberInfo.isAuthenticated ?
+										<a  href="/logout">Sign out</a>
```

Comment:
I could add the action, but we don't actually need it with our awesome routing :P ?